### ISSUE TITLE
Setting wifi credentials with platformio terminal

### ISSUE THREAD

**Issue by aaaidan** [HomeSpan/HomeSpan#9]
### Issue: can't set wifi password on PlatformIO

I was trying to set the wifi password in PlatformIO's serial monitor a few times, but it never worked (it failed to connect as if I had typed the wrong password). Then I remembered I'd succeeded in Arduino's serial monitor, so I tried it there and it worked first try (same password).

PIO's serial monitor seems to work well with all other aspects of the Homespan serial UI, so I am guessing it is just sending commands slightly differently (perhaps an extra control character or a different line ending).

As at #3, @HomeSpan hadn't decided whether platformio is a "supported" target of the library, so this issue may be out of scope for the project.  So ...

### Aside: HomeSpan's "support" of PIO

Aside from this issue, it would be helpful to know whether PIO is in or out (even if just for now), because I don't want to clutter the project with unwelcome bugs and requests. PIO offers the Arduino Core as a framework for the ESP32 platform, which explains why HomeSpan seems to build and run perfectly. So perhaps the support burden is low enough to justify targeting it? 🤷‍♂️ 

Fwiw, for projects with any significant complexity, I think PIO is a _much_ better IDE than Arduino. 

Let me know if you want this question (of support) as a separate issue.

**Comment by chandime** [comment]
I am having the same issue. Hope someone could provide a solution.

**Comment by HomeSpan** [comment]
Sorry for the delay on this.  I don't have PIO, but based on the symptoms I think I figured out the problem.  My guess is that Platform IO is sending a carriage return & newline to terminate each serial line.  If so, this would explain why all commands seem to work except for setting the WiFi Credentials (all other commands only look at the first character).

I've modified my code to strip out any embedded '\r' characters and checked that this works with the Arduino Serial Monitor even if you change the settings to send Both CR & NL at the end of each line.

The new code has been updated in the ***development*** branch.  Can you give it a try and let me know if it solves the problem?  If so, I'll include in the next full patch release.

Thanks!

**Comment by chandime** [comment]
The issue wasn't solved. Since it was an issue with the PIO Terminal, I switched to Arduino IDE. Thanks anyways!

**Comment by aaaidan** [comment]
That’s a shame. 

I haven’t checked, myself, but if it really does persist, I will take a closer look at the way PIO and Arduino differ with the way they send commands and let you know what I find

**Comment by aaaidan** [comment]
Actually, I just pulled down Development to check if it was fixed, and it looks like it is. I'm able to set a wifi password that works. 👍 

Not sure what happened for @chandime. I had a whale of a time figuring out how to install the branch in question, so maybe they were accidentally testing v1.1.3?

For future reference, installing the git repo url directly (not the `homespan/HomeSpan` package name) is the way to go. Like this:

    [env:nodemcu32]
    platform = espressif32
    board = nodemcu-32s
    framework = arduino
    upload_port = /dev/cu.SLAB_USBtoUART
    monitor_port = /dev/cu.SLAB_USBtoUART
    monitor_speed = 115200
    lib_deps = git@github.com:HomeSpan/HomeSpan.git#Development

Easiest way is to run `pio lib install git@github.com:HomeSpan/HomeSpan.git#Development` in the PIO Core CLI. It should save it to your `platformio.ini` by default.
